File: connect/src/main/java/org/transdroid/connect/clients/UnsupportedFeatureException.java
package org.transdroid.connect.clients;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;

/**
 * Thrown when trying to call into a client method for a feature which the client does not support.
 */
public class UnsupportedFeatureException extends NotImplementedException {

	private final Client client;
	private final Feature feature;

	UnsupportedFeatureException(Client client, Feature feature) {
		this.client = client;
		this.feature = feature;
	}

	public String getMessage() {
		return client.name() + " does not support " + feature.name();
	}

}

File: connect/src/main/java/org/transdroid/connect/clients/UnsupportedFeatureException.kt
package org.transdroid.connect.clients

/**
 * Thrown when trying to call into a client method for a feature which the client does not support.
 */
class UnsupportedFeatureException internal constructor(
        private val client: Client,
        private val feature: Feature) : RuntimeException() {

    override val message: String?
        get() = client.name + " does not support " + feature.name

}
